staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -187,6 +189,20 @@ func New(
 		time.Duration(0),
 	)
 
+	if enableDirectConnect {
+		endpointsInformer.Informer().AddEventHandlerWithResyncPeriod(
```

Comment:
service controller and the associated provider interface is designed specifically for nodes as the backend, I would recommend writing a new controller for this use-case

Reply:
I think a new controller is better too. do u mean an endpoints controller? it watches the endpoints, and only reconciles backends of the load balancer.